apisix/init.lua

```diff
@@ -629,6 +630,12 @@ end
 function _M.http_header_filter_phase()
     core.response.set_header("Server", ver_header)
 
+    local status_code = tonumber(get_var("upstream_status"))
+    if status_code and status_code >= 500 and status_code <= 599 then
+        core.response.set_header("X-APISIX-Upstream-Status", status_code)
```

Comment:
How about `X-Upstream-Status`?

Reply:
I think the `X-APISIX-` prefix is required to distinguish headers source, it's confused if backend also carries headers with `X-Upstream-` prefix.

Reply:
> I think the `X-APISIX-` prefix is required to distinguish headers source, it's confused if backend also carries headers with `X-Upstream-` prefix.

agree +1